### PR TITLE
Fix for new clippy warnings

### PR DESCRIPTION
**Pull request by Fraser999**
This fixes clippy warnings introduced by Rust 1.62.0.  It also adds missing features to the `casper-types` entry of the `casper-hashing` manifest.

### CODE REVIEW

File: hashing/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ license = "Apache-2.0"
 [dependencies]
 blake2 = "0.9.0"
 base16 = "0.2.1"
-casper-types = { version = "1.5.0", path = "../types" }
+casper-types = { version = "1.5.0", path = "../types", features = ["datasize", "std"] }
```
Comment (goral09) [inline hashing/Cargo.toml:15]: What changed that now we need to add these features just now? Is it something in our software or in the rustc?

File: node/src/testing/network.rs [hunk 2]
```diff
@@ -74,9 +74,9 @@ where
     ///
     /// Panics if a duplicate node ID is being inserted. This should only happen in case a randomly
     /// generated ID collides.
-    pub(crate) async fn add_node(
-        &mut self,
-        rng: &mut TestRng,
+    pub(crate) async fn add_node<'a, 'b: 'a>(
```
Comment (goral09) [inline node/src/testing/network.rs:77]: Does it mean that `'b` outlives `'a`?

**Review by goral09 — APPROVED** [review]
How can I argue with clippy and rustc 🤷

File: hashing/Cargo.toml (hunk 1 above)
Comment (Fraser999) [reply hashing/Cargo.toml:15]: We probably just never test `casper-hashing` in isolation.  When we run `cargo t` from the workspace root, `casper-hashing` will get compiled using a build of `casper-types` which has the required features enabled due to other crates (like `casper-node`) demanding these features to be enabled.

File: node/src/testing/network.rs (hunk 2 above)
Comment (Fraser999) [reply node/src/testing/network.rs:77]: Yes

**Comment by Fraser999** [comment]
bors merge

**Review by mpapierski — APPROVED** [review]

**Pull request merged**